### PR TITLE
Adding in memory table

### PR DESCRIPTION
**Pull request by Alexzerntev**
- Added in memory table that supports CRUD
- Added initial readme

### CODE REVIEW

File: src/main/scala/com/rawlabs/das/mock/DASMockStorage.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  def update(rowId: String, row: Row): Row = {
+    treeMap.put(rowId, row)
+    row
+  }
+
+  def remove(rowId: String): Unit = treeMap.remove(rowId)
+
+  def size: Int = treeMap.size
+
+  def iterator: Iterator[Row] = treeMap.valuesIterator
```
Comment (miguelbranco80) [inline src/main/scala/com/rawlabs/das/mock/DASMockStorage.scala:37]: Interesting case. How to give back an iterator if the underlying structure can change?
Clone the treeMap and give an iterator out of the cloned one.

File: src/main/scala/com/rawlabs/das/mock/DASMockStorage.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package com.rawlabs.das.mock
+
+import com.rawlabs.protocol.das.Row
+
+import scala.collection.mutable;
+
+class DASMockStorage(
+    val key: String,
+    private val treeMap: mutable.TreeMap[String, Row] = new mutable.TreeMap[String, Row]()
```
Comment (miguelbranco80) [inline src/main/scala/com/rawlabs/das/mock/DASMockStorage.scala:21]: Move this to the body of the class.
Also, create a lock object.

```scala
class DASMockStorage(val key: String) {

  private val treeMap = new mutable.TreeMap[String, Row]()
  private val treeMapLock = new Object

  def add(row: Row): Row = {
    treeMapLock.synchronized {
      treeMap.put(row.getDataMap.get(key).getInt.getV.toString, row)
      row
    }
  }
  ... <same elsewhere>

```

File: src/main/scala/com/rawlabs/das/mock/DASMockTable.scala [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+  override def explain(
+      quals: Seq[Qual],
+      columns: Seq[String],
+      maybeSortKeys: Option[Seq[SortKey]],
+      maybeLimit: Option[Long],
+      verbose: Boolean
+  ): Seq[String] = Seq.empty
+
+  override def execute(
```
Comment (miguelbranco80) [inline src/main/scala/com/rawlabs/das/mock/DASMockTable.scala:38]: Add a comment saying quals and columsn are being ignored (PostgreSQL does it anyway).
But maybeLimit is being applied.

File: src/main/scala/com/rawlabs/das/mock/DASMockInMemoryTable.scala [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+  override def explain(
+      quals: Seq[Qual],
+      columns: Seq[String],
+      maybeSortKeys: Option[Seq[SortKey]],
+      maybeLimit: Option[Long],
+      verbose: Boolean
+  ): Seq[String] = Seq.empty
+
+  override def execute(
```
Comment (miguelbranco80) [inline src/main/scala/com/rawlabs/das/mock/DASMockInMemoryTable.scala:41]: Add a comment saying quals, columns and maybeLimit are being ignored.

File: src/main/scala/com/rawlabs/das/mock/DASMockInMemoryTable.scala [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+      quals: Seq[Qual],
+      columns: Seq[String],
+      maybeSortKeys: Option[Seq[SortKey]],
+      maybeLimit: Option[Long]
+  ): DASExecuteResult = {
+    logger.info(s"Executing query with quals: $quals, columns: $columns, sortKeys: $maybeSortKeys, limit: $maybeLimit")
+
+    new DASExecuteResult {
```
Comment (miguelbranco80) [inline src/main/scala/com/rawlabs/das/mock/DASMockInMemoryTable.scala:49]: This seems wrong. If you say you support sort orders, you must implement sort orders here.
Did you try a query with various ORDER BY ASC and DESC and see if it behaves? I suspect it's broken.

File: README.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+- **in_memory** is a table that stores data in memory.
+
+## Running the service in IntelliJ
+
+After setting up the environment and building the project, go to `External Libraries` ->
+`sbt: com.raw-labs:das-server-scala_<version>:jar`.
+Open DASSServerMain and run the main method.
+
+## Interact with the DASMock FDW
+```DROP SERVER multicorn_das CASCADE;
```
Comment (miguelbranco80) [inline README.md:28]:
```sql

File: README.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
+# DAS Mock
+
+This is a mock implementation for the DAS service. It demonstrates how to build create a simple DAS service and its
+underlying data model.
+
+## Prerequisites
+
+- PostgreSQL 13+
+- [multicorn2](https://github.com/raw-labs/multicorn2)
```
Comment (miguelbranco80) [inline README.md:9]: multicorn2 RAW Labs Fork

File: src/main/scala/com/rawlabs/das/mock/DASMock.scala [hunk 8]
```diff
@@ -13,11 +13,19 @@
...
 import com.typesafe.scalalogging.StrictLogging
 
 class DASMock extends DASSdk with StrictLogging {
+
+  private val dasMockStorage = new DASMockStorage("column1")
+
+  def this(options: Map[String, String]) = {
+    this()
+    options.keys.foreach(key => logger.info(s"Option: $key = ${options(key)}"))
+  }
```
Comment (miguelbranco80) [inline src/main/scala/com/rawlabs/das/mock/DASMock.scala:27]: ?

Comment (miguelbranco80) [reply src/main/scala/com/rawlabs/das/mock/DASMock.scala:27]: Why don't you add this to the class constructor?

**Review by miguelbranco80 — CHANGES_REQUESTED** [review]
Some suggestions

**Review by bgaidioz — APPROVED** [review]

**Review by miguelbranco80 — APPROVED** [review]

**Pull request merged**